app/controllers/v0/terms_of_use_agreements_controller.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module V0
+  class TermsOfUseAgreementsController < ApplicationController
+    before_action :set_user_account
+    before_action :set_terms_of_use_agreement
+
+    STATSD_PREFIX = 'api.terms_of_use_agreements'
+
+    def accept
+      @terms_of_use_agreement.accepted!
+      render_success
+    rescue ActiveRecord::RecordInvalid
+      render_error
+    end
+
+    def decline
+      @terms_of_use_agreement.declined!
+      render_success
+    rescue ActiveRecord::RecordInvalid
+      render_error
+    end
+
+    private
+
+    def set_user_account
+      @user_account = current_user.user_account
+    end
+
+    def set_terms_of_use_agreement
+      @terms_of_use_agreement = @user_account.terms_of_use_agreements.new(agreement_version: params[:version])
+    end
+
+    def render_success
+      render json: { terms_of_use_agreement: @terms_of_use_agreement }, status: :created
+      log_success
+    end
+
+    def render_error
+      render json: { error: @terms_of_use_agreement.errors.full_messages.join(', ') }, status: :unprocessable_entity
+    end
+
+    def log_success
+      context = {
+        terms_of_use_agreement_id: @terms_of_use_agreement.id,
+        user_account_id: @user_account.id,
```

Comment:
Should this match the UUID?

Reply:
Yeah, the `id` is a `uuid` type. Is that what you're asking?

Reply:
Ah gotcha. No it shouldn't. `user.uuid` is the CSPs `uuid`. This will match `user.user_account_uuid`. I'll change it to `user_account_uuid` to make it clearer